docs/sources/whatsnew/whats-new-in-v9-2.md

```diff
@@ -219,3 +219,17 @@ For more information, see the [SAML configuration documentation](https://grafana
 You can now map OAuth groups and roles to Server Admin for the GitLab, GitHub, AzureAD, Okta, and Generic OAuth integrations.
 To enable this functionality, set the `allow_assign_grafana_admin` configuration option to `true` in the desired OAuth integration section.
 For more information, see the [authentication configuration documentation](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/) for each OAuth client.
+
+## Match parameter support in prometheus labels API
+
+Prometheus users running Prometheus 2.24 or higher can now elect to use the
+[labels](https://prometheus.io/docs/prometheus/latest/querying/api/#querying-label-values) endpoint instead of the
+[series](https://prometheus.io/docs/prometheus/latest/querying/api/#finding-series-by-label-matchers) endpoint for the
+`label_values` [function](https://grafana.com/docs/grafana/latest/datasources/prometheus/#query-variable).
+This has no impact on dashboard functionality, but can decrease load times for templated high cardinality
+prometheus instances.
+
+In order to prevent breaking changes for prometheus versions before 2.24, users must first configure their prometheus
+type and version in the prometheus
+[configuration](https://grafana.com/docs/grafana/latest/administration/data-source-management),
+or in the [provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/) file.
```

Comment:
When should Prometheus <v2.24 users "first configure their prometheus type and version"? Before upgrading to v9.2? Immediately after? What breaks if they don't?

cf. my comment in #56496 - Prometheus-specific settings, including provisioning, should also be in the [data source's docs](https://github.com/grafana/grafana/blob/main/docs/sources/datasources/prometheus.md), which should also make it a more specific link target for configuration docs.

```suggestion
Prometheus users running Prometheus v2.24 and higher can use the [labels endpoint](https://prometheus.io/docs/prometheus/latest/querying/api/#querying-label-values) instead of the [series endpoint](https://prometheus.io/docs/prometheus/latest/querying/api/#finding-series-by-label-matchers) for the [`label_values` function]({{< relref "../datasources/prometheus/#query-variable" >}}).
This decreases load times for templated high-cardinality Prometheus instances.

To prevent breaking changes if you're running Prometheus versions before v2.24, you must first configure the Prometheus type and version in any Prometheus data sources' [configuration]({{< relref "../datasources/prometheus/" >}}).
```

Reply:
> When should Prometheus <v2.24 users "first configure their prometheus type and version"? Before upgrading to v9.2? Immediately after? What breaks if they don't?

If you're running prometheus <2.24, the new API endpoint we're exposing in this PR doesn't exist, so they'll continue calling the existing series API that we call today, nothing breaks, they just can't use these new optimizations we're releasing.

If you're running prometheus <2.24 you don't need to set your configuration at all, as you don't apply for the new functionality.

Only users that are running prometheus >2.24 benefit from setting their configuration, as then they will be able to use new API endpoints that will increase performance.

Reply:
I'm confused by "In order to prevent breaking changes for prometheus versions before 2.24" then - it doesn't sound like there are any breaking changes or necessary configuration changes regardless of version?

If there's not a change that will break things when upgrading to Grafana v9.2 then we need to avoid the term "breaking change".

Reply:
If you want to merge this I'm OK to iterate on the docs end. But I really want to avoid suggesting this might be a breaking change if there's not one.

Reply:
Oh yeah, that is unnecessarily confusing, I was trying to explain why we're releasing this functionality as opt-in, because if we otherwise assume that everyone supports the new API unless they configure an older version, then folks running older versions could see dashboards break. I'll rewrite this, I don't think it's super relevant anyway

Reply:
I changed it to:
`If you want to benefit from this endpoint you must first configure the Prometheus type and version in any Prometheus data sources' [configuration]({{< relref "../datasources/prometheus/" >}}).`